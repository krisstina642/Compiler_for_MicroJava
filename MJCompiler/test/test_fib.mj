//Test 302

program test302

const int jedan = 1;

int i, j;
bool ibool;

int niz[];

{
	
	int sum(int z, int d)
	{
		return z + d;
	}
	
	void fibonaci(int a[]) 
		int pom,l; 
	{
		pom=len(a);
		
		if (pom<1) return;
		l=2;
		a[0]=0;
		if (pom<2) return;
		a[1]=1;
		
		if(pom<3) return;
		
		do
		{
			a[l]=sum(a[l-1],a[l-2]);
			l++;
		}
		while( l < pom );
		
	}
	
	void main()
	{
				
		niz = new int[7];
		fibonaci(niz);
		i=0;
		
		do
		{
			print(niz[i]);
			i++;
		}
		while( i < len(niz));
		
		print(eol);
		print(i);
	}
}

