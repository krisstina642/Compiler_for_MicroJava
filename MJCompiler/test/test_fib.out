codeSize=189
dataSize=4
mainPC=113
0: enter 2 2
3: load_0
4: load_1
5: add
6: exit
7: return
8: exit
9: return
10: enter 1 3
13: load_0
14: arraylength
15: store_1
16: load_1
17: const_1
18: jge 7 (=25)
21: const_1
22: jmp 4 (=26)
25: const_0
26: const_0
27: jeq 5 (=32)
30: exit
31: return
32: const_2
33: store_2
34: load_0
35: const_0
36: const_0
37: astore
38: load_1
39: const_2
40: jge 7 (=47)
43: const_1
44: jmp 4 (=48)
47: const_0
48: const_0
49: jeq 5 (=54)
52: exit
53: return
54: load_0
55: const_1
56: const_1
57: astore
58: load_1
59: const_3
60: jge 7 (=67)
63: const_1
64: jmp 4 (=68)
67: const_0
68: const_0
69: jeq 5 (=74)
72: exit
73: return
74: load_0
75: load_2
76: load_0
77: load_2
78: const_1
79: sub
80: aload
81: load_0
82: load_2
83: const_2
84: sub
85: aload
86: call -86 (=0)
89: astore
90: load_2
91: const_1
92: add
93: store_2
94: load_2
95: load_1
96: jge 7 (=103)
99: const_1
100: jmp 4 (=104)
103: const_0
104: const_0
105: jeq 6 (=111)
108: jmp -34 (=74)
111: exit
112: return
113: enter 0 0
116: const 7
121: newarray 1
123: putstatic 3
126: getstatic 3
129: call -119 (=10)
132: const_0
133: putstatic 0
136: getstatic 3
139: getstatic 0
142: aload
143: const_5
144: print
145: getstatic 0
148: const_1
149: add
150: putstatic 0
153: getstatic 0
156: getstatic 3
159: arraylength
160: jge 7 (=167)
163: const_1
164: jmp 4 (=168)
167: const_0
168: const_0
169: jeq 6 (=175)
172: jmp -36 (=136)
175: const 10
180: const_1
181: bprint
182: getstatic 0
185: const_5
186: print
187: exit
188: return
    0    1    1    2    3    5    8
    7
Completion took 0 ms